services/tinlake/actions.ts

```diff
@@ -313,205 +300,151 @@ export async function borrow(tinlake: any, loan: Loan, amount: string) {
       }
     } catch (e) {
       return loggedError(e, `Could not approve currency for ${trancheType}.`, '')
     }
-    if (approveRes.status !== SUCCESS_STATUS) {
-      return loggedError({}, `Could not approve currency for ${trancheType}.`, '')
-    }
   }
 
   // supply
-  let supplyRes
   try {
     if (trancheType === 'junior') {
-      supplyRes = await tinlake.supplyJunior(supplyAmount)
-    } else if (trancheType === 'senior') {
-      supplyRes = await tinlake.supplySenior(supplyAmount)
+      return await tinlake.supplyJunior(supplyAmount)
     }
+    return await tinlake.supplySenior(supplyAmount)
   } catch (e) {
     return loggedError(e, `Could not supply ${trancheType}`, '')
   }
-  if (supplyRes.status !== SUCCESS_STATUS) {
-    return loggedError({}, `Could not supply ${trancheType}`, '')
-  }
-
-  return supplyRes
 }
 
-export async function redeem(tinlake: ITinlake, redeemAmount: string, trancheType: TrancheType) {
+export async function redeem(
+  tinlake: ITinlake,
+  redeemAmount: string,
+  trancheType: TrancheType
+): Promise<PendingTransaction> {
+  const address = await tinlake.signer?.getAddress()
+
   let allowance = new BN(0)
   if (trancheType === 'junior') {
-    allowance = (await tinlake.getJuniorTokenAllowance((tinlake.ethConfig as any).from!)) || new BN(0)
+    allowance = (await tinlake.getJuniorTokenAllowance(address!)) || new BN(0)
   } else if (trancheType === 'senior') {
-    allowance = (await tinlake.getSeniorTokenAllowance((tinlake.ethConfig as any).from!)) || new BN(0)
+    allowance = (await tinlake.getSeniorTokenAllowance(address!)) || new BN(0)
   }
 
   // only approve if allowance is smaller than than redeemAmount
   if (allowance.lt(new BN(redeemAmount))) {
     // approve junior token
-    let approveRes: any
     try {
       if (trancheType === 'junior') {
-        approveRes = await tinlake.approveJuniorToken(maxUint256)
+        await tinlake.approveJuniorToken(maxUint256)
       } else if (trancheType === 'senior') {
-        approveRes = await tinlake.approveSeniorToken(maxUint256)
+        await tinlake.approveSeniorToken(maxUint256)
       }
     } catch (e) {
       return loggedError(e, `Could not approve ${trancheType} Token.`, '')
     }
-    if (approveRes.status !== SUCCESS_STATUS) {
-      return loggedError({}, `Could not approve ${trancheType} Token.`, '')
-    }
   }
 
   // repay
-  let redeemRes
   try {
     if (trancheType === 'junior') {
-      redeemRes = await tinlake.redeemJunior(redeemAmount)
-    } else if (trancheType === 'senior') {
-      redeemRes = await tinlake.redeemSenior(redeemAmount)
+      return await tinlake.redeemJunior(redeemAmount)
+    }
+    if (trancheType === 'senior') {
+      return await tinlake.redeemSenior(redeemAmount)
     }
   } catch (e) {
     return loggedError(e, `Could not redeem ${trancheType}.`, '')
   }
-  if (redeemRes.status !== SUCCESS_STATUS) {
-    return loggedError({}, `Could not redeem ${trancheType}.`, '')
-  }
```

Comment:
Don't we need the status checks anymore? Is that happening within tinlake.js?

Reply:
We check the status in the transaction handler in the `transactions` duck, so this was just duplicate (and returning a basically useless error message, while Tinlake.js can/does often return better errors).